### DISCUSSION TITLE
Inconsistent site template application issues using Azure RunBook

### DISCUSSION THREAD

**Discussion by diver7** [pnp/powershell#461] [Q&A]
I have an Azure Automation Powershell RunBook that we use to create SharePoint Team sites. Part of that script applies a couple of templates to the newly created site using Invoke-PnPSiteTemplate. I am seeing intermittent errors when applying the template. I can run it one time and it completes without any issues. I can then run it again for another site and it fails when applying the template with this error (GUIDs have been replaced):

`Invoke-PnPSiteTemplate : The object id "site:00000000-0000-0000-0000-000000000000:web:00000000-0000-0000-0000-000000000000 75:contenttype:0x0120D52000483357DD5D6C6A42B9B0471DCC5B5DEA:fl:00000000-0000-0000-0000-000000000000" is invalid. At line:87 char:9 + Invoke-PnPSiteTemplate -Path https://wipflillp.sharepoint.com ... + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ + CategoryInfo : WriteError: (:) [Invoke-PnPSiteTemplate], ServerException + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Provisioning.Site.InvokeSiteTemplate`  

These are brand new sites being created in the script right above the template being applied. The template files are being stored and read from a SharePoint library and are not changing. I'm not sure how something could become "invalid" from one run to the next. Once the runbook finishes I can run the site template command locally and it completes without issue. I have tried PnP.Powershell v1.2 and v1.4 with the same results. Can anyone provide some ideas on what could be causing this?

**Comment by diver7** [comment]
Here's a new error: 
`Invoke-PnPSiteTemplate : The parent content type specified by content type identifier 0x0120D520 does not exist. At line:87 char:9 + Invoke-PnPSiteTemplate -Path https://tenant.sharepoint.com ... + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ + CategoryInfo : WriteError: (:) [Invoke-PnPSiteTemplate], ServerException + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Provisioning.Site.InvokeSiteTemplate`

This one also happened during applying the template. Seems to be a possible communication error. Maybe it's not getting a response.

**Reply by stefwanders** [reply]
Have you considered maybe timing your actions in the script? I qoute you "These are brand new sites being created in the script right above the template being applied". I don't know how your script looks like, but often when I create a large amount of groups myself and apply the Template, I get errors as well, simply because the script runs too fast and in the back the SharePoint isnt fully setup yet.

Also, maybe this will help? Add-PnPSiteDesignTask
https://pnp.github.io/powershell/cmdlets/Add-PnPSiteDesignTask.html

"This command is used to apply a published site design to a specified site collection target. It schedules the operation, allowing for the application of larger site scripts (Invoke-PnPSiteDesign is limited to 30 actions and subactions). This command is intended to replace Invoke-PnPSiteDesign and is useful when you need to apply a large number of actions or multiple site scripts."

-note; I havent myself used the Add-PnPSiteDesignTask

**Reply by diver7** [reply]
@stefwanders Thanks for the ideas. I already have pauses built into the script because you have to wait 3 minutes after creating a site before you can apply a template. Otherwise it will fail every time. This issue doesn't appear to be a specific timing issue. My script creates the site, waits 3 minutes and then applies the template. Most of the time it will run just fine without and errors. Then I will have errors multiple times in a row. The errors change too. This is why I think it may be something with Azure communicating with SPO. Azure tried to read the template file and only gets part of it, maybe. I would like to try storing the template file somewhere in Azure but I'm not sure where I could store it.
